src/Our.Umbraco.Ditto/Extensions/PublishedContentExtensions.cs

```diff
@@ -216,6 +224,294 @@ public static class PublishedContentExtensions
         }
 
         /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="type">
+        /// The <see cref="Type"/> to convert too.
+        /// </param>
+        /// <param name="instance">
+        /// The <see cref="object"/> to populate.
+        /// </param>
+        /// <param name="propertyValue">
+        /// The <see cref="object"/> whos property we will populate with.
+        /// </param>
+        /// <param name="actualPropertyName">
+        /// The actual <see cref="string"/> name of the property.
+        /// </param>
+        /// <param name="propertyInfo"></param>
+        /// The <see cref="PropertyInfo"/> of the property to set on the object
+        /// <param name="content">
+        /// The <see cref="IPublishedContent"/> of which to use as a base for populating content
+        /// </param>
+        /// <param name="isArchetype">
+        /// A <see cref="bool"/> specifying if is of type archetype
+        /// </param>
+        private static void SetValue(Type type, object instance, object propertyValue, PropertyInfo propertyInfo, IPublishedContent content = null, bool isArchetype = false)
+        {
+            if (propertyValue != null)
+            {
+                var propertyType = propertyInfo.PropertyType;
+                var typeInfo = propertyType.GetTypeInfo();
+                var isEnumerableType = propertyType.IsEnumerableType() &&
+                                       typeInfo.GenericTypeArguments.Any();
+
+                // Try any custom type converters first.
+                // 1: Check the property.
+                // 2: Check any type arguments in generic enumerable types.
+                // 3: Check the type itself.
+
+                var converterAttribute =
+                    propertyInfo.GetCustomAttribute<TypeConverterAttribute>()
+                    ??
+                    (isEnumerableType
+                         ? typeInfo.GenericTypeArguments.First()
+                                   .GetCustomAttribute<TypeConverterAttribute>(true)
+                         : propertyType.GetCustomAttribute<TypeConverterAttribute>(true));
+
+                if (isArchetype)
+                {
+                    /* [ML]: TODO - There must be a better way to do this to avoid an archetype dependency with its model binding to CMS json. */
+
+                    var json = propertyValue is string ? propertyValue as string : JsonConvert.SerializeObject(propertyValue, new JsonSerializerSettings
+                    {
+                        ReferenceLoopHandling = ReferenceLoopHandling.Ignore
+                    });
+
+                    var jsonModel = JsonConvert.DeserializeObject<ArchetypeCMSModel>(json);
```

Comment:
@leekelleher I deserialized and reserialized to my own class to avoid a dependency on the Archetype dll. Umbraco seems to find the ArchetypeModel within the Archetype.dll as the property value so didn't want to pass this about.